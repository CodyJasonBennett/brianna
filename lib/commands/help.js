const { readdirSync } = require('fs');
const { join } = require('path');
const { bold, white } = require('../utils/colors');
const { createTable } = require('../utils/table');

const COMMANDS_DIR = join(__dirname, '../commands');

const help = {
  aliases: ['-h', 'help'],
  description: "Displays Brianna's commands and options.",
  flags: [
    {
      aliases: ['<command>'],
      description: 'Name of command to display additional info.',
    },
  ],
  run: args => {
    const [query] = args.slice(1);

    if (query) {
      const command = readdirSync(COMMANDS_DIR)
        .map(name => name.replace('.js', ''))
        .find(name => name === query);
      if (!command) throw new Error(`Command "${query}" not found.`);

      const { description, flags } = require(join(COMMANDS_DIR, command));

      const usage = `\`Usage: ${bold(
        white(`brianna ${command}${flags ? ' [flags]' : ''}`)
      )}\``;

      const options = flags ? `\n\n${createTable(flags, 'Options:')}` : '';

      const info = `Visit ${bold(
        white(`https://briannacli.com/docs/${query}`)
      )} to learn more about this command.`;

      console.info(`${usage}\n\n${description}${options}\n\n${info}`);
    } else {
      const usage = `Usage: \`${bold(white('brianna <command> [flags]'))}\``;

      const commands = createTable(
        readdirSync(COMMANDS_DIR).map(name => require(join(COMMANDS_DIR, name))),
        'Commands:'
      );

      const instructions = `Run \`${bold(
        white('brianna help <command>')
      )}\` for more information on specific commands.`;

      const info = `Visit ${bold(
        white('https://briannacli.com')
      )} to learn more about Brianna.`;

      console.info(`${usage}\n\n${commands}\n\n${instructions}\n${info}`);
    }
  },
};

module.exports = help;
