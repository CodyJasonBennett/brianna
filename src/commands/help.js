import { readdirSync } from 'fs';
import { join } from 'path';
import { createTable } from 'utils/table';
import colors from 'utils/colors';

const COMMANDS_DIR = join(__dirname, '../commands');

const help = {
  aliases: ['-h', 'help'],
  description: "Displays Brianna's commands and options.",
  flags: [
    {
      aliases: ['<command>'],
      description: 'Name of command to display additional info.',
    },
  ],
  run: (...args) => {
    const [query] = args;

    if (query) {
      const command = readdirSync(COMMANDS_DIR)
        .map(name => name.replace('.js', ''))
        .find(name => name === query);
      if (!command) throw new Error(`Command "${query}" not found.`);

      const { description, flags } = require(join(COMMANDS_DIR, command)).default;

      const usage = `\`Usage: ${colors.bold(
        colors.white(`brianna ${command}${flags ? ' [flags]' : ''}`)
      )}\``;

      const options = flags
        ? `\n\nOptions:${createTable(flags, 'aliases', 'description')}`
        : '';

      const info = `Visit ${colors.bold(
        colors.white(`https://briannacli.com/docs/${query}`)
      )} to learn more about this command.`;

      console.info(`${usage}\n\n${description}${options}\n\n${info}`);
    } else {
      const usage = `Usage: \`${colors.bold(
        colors.white('brianna <command> [flags]')
      )}\``;

      const commands = createTable(
        readdirSync(COMMANDS_DIR).map(name => require(join(COMMANDS_DIR, name)).default),
        'aliases',
        'description'
      );

      const instructions = `Run \`${colors.bold(
        colors.white('brianna help <command>')
      )}\` for more information on specific commands.`;

      const info = `Visit ${colors.bold(
        colors.white('https://briannacli.com')
      )} to learn more about Brianna.`;

      console.info(`${usage}\n\nCommands:${commands}\n\n${instructions}\n${info}`);
    }
  },
};

export default help;
