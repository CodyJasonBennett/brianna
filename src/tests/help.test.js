import help from 'commands/help';

describe('commands/help', () => {
  it('displays info with no arguments', () => {
    let output;
    console.info = console.error = args => (output = args);

    help.run();

    expect(output).toMatchSnapshot();
  });

  it('displays command info', () => {
    let output;
    console.info = console.error = args => (output = args);

    help.run(null, 'install');

    expect(output).toMatchSnapshot();
  });
});
