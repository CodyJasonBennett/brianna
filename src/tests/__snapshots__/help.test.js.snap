// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/help displays command info 1`] = `
"\`Usage: [1m[37mbrianna install [flags][39m[22m\`

Installs all dependencies for a project.

Options:
  -P, --production  install production dependencies            
  -D, --save-dev    install package to your \`devDependencies\`  

Visit [1m[37mhttps://briannacli.com/docs/install[39m[22m to learn more about this command."
`;

exports[`commands/help displays info with no arguments 1`] = `
"Usage: \`[1m[37mbrianna <command> [flags][39m[22m\`

Commands:
  -h, help     Displays Brianna's commands and options.  
  -i, install  Installs all dependencies for a project.  
  -v, version  Get Brianna's current version.            

Run \`[1m[37mbrianna help <command>[39m[22m\` for more information on specific commands.
Visit [1m[37mhttps://briannacli.com[39m[22m to learn more about Brianna."
`;
